### PR TITLE
builders: fix _is_arxiv regex bug

### PR DESCRIPTION
**Pull request by vbalbp**
Signed-off-by: Victor Balbuena <vbalbp@gmail.com>

### CODE REVIEW

File: inspire_schemas/builders/references.py [hunk 1]
```diff
@@ -45,7 +45,17 @@
 # Matches new style arXiv ID, with an old-style class specification
 # (Malformed, but appears in APS records)
 RE_ARXIV_POST_2007_CLASS = re.compile(
-    "(arxiv:)?(?:[a-z\-]+)(?:\.[a-z]{2})?/(\d{4})\.(\d{4,5})(v\d+)?$",
+    "(arxiv:)?([a-z\-]+)(?:\.[a-z]{2})?/(\d{4})\.(\d{4,5})(v\d+)?$",
```
Comment (michamos) [inline inspire_schemas/builders/references.py]: I don't think this is correct, you should match the whole textual part for the category.
It's also easier to use identifiers. Like
```
"(arxiv:)?((?P<category>(?:[a-z\-]+)(?:\.[a-z]{2})?)/)?(P<identifier>\d{4})\.(\d{4,5})(v\d+)?$"

File: inspire_schemas/builders/references.py [hunk 2]
```diff
@@ -45,7 +45,17 @@
...
+    flags=re.I
+)
+
+RE_ARXIV_PRE_2007_CLASS = re.compile(
+    "(arxiv:)?([a-z\-]+)(\.[a-z]{2})?(/\d{4})(\d+)(v\d+)?$",
+    flags=re.I
+)
+
+RE_ARXIV_POST_2007 = re.compile(
```
Comment (michamos) [inline inspire_schemas/builders/references.py]: You can remove this, as the other pattern will also match this.

File: inspire_schemas/builders/references.py [hunk 3]
```diff
@@ -45,7 +45,17 @@
 # Matches new style arXiv ID, with an old-style class specification
 # (Malformed, but appears in APS records)
 RE_ARXIV_POST_2007_CLASS = re.compile(
-    "(arxiv:)?(?:[a-z\-]+)(?:\.[a-z]{2})?/(\d{4})\.(\d{4,5})(v\d+)?$",
+    "(arxiv:)?([a-z\-]+)(?:\.[a-z]{2})?/(\d{4})\.(\d{4,5})(v\d+)?$",
+    flags=re.I
+)
+
+RE_ARXIV_PRE_2007_CLASS = re.compile(
```
Comment (michamos) [inline inspire_schemas/builders/references.py]: same idea.

File: inspire_schemas/builders/references.py [hunk 4]
```diff
@@ -114,9 +124,20 @@ def _is_arxiv(obj):
...
+    matched_non_class = RE_ARXIV_POST_2007.match(arxiv_test[0])
+
+    if matched_non_class:
+        return True
+
+    matched_arxiv = RE_ARXIV_PRE_2007_CLASS.match(arxiv_test[0]) \
         or RE_ARXIV_POST_2007_CLASS.match(arxiv_test[0])
 
+    if matched_arxiv:
+        return matched_arxiv.group(2) in valid_arxiv_categories()
```
Comment (michamos) [inline inspire_schemas/builders/references.py]: you should rather do something like this to account for casing and variations: https://github.com/inspirehep/inspire-schemas/blob/8ad58e4f18f42049f089f959dc7b8f2eda73c87a/inspire_schemas/utils.py#L402-L405

File: inspire_schemas/utils.py [hunk 5]
```diff
@@ -957,3 +969,41 @@ def normalize_isbn(isbn):
...
+    modified here to work with two regular expressions instead
+    of three and adding a check for valid arxiv categories only"""
+    arxiv_test = obj.split()
+    if not arxiv_test:
+        return False
+
+    matched_arxiv = (RE_ARXIV_PRE_2007_CLASS.match(arxiv_test[0]) or
+                     RE_ARXIV_POST_2007_CLASS.match(arxiv_test[0]))
+
+    if matched_arxiv:
```
Comment (michamos) [inline inspire_schemas/utils.py]: it's clearer to do it the other way around:
```
if not matched_arxiv:
    return False

# do the rest
```

File: inspire_schemas/utils.py [hunk 6]
```diff
@@ -957,3 +969,41 @@ def normalize_isbn(isbn):
...
+        valid_arxiv_categories_lower = [category.lower() for category in valid_arxiv_categories()]
```
Comment (michamos) [inline inspire_schemas/utils.py]: you only need this if you have matched a category, so this can go inside the `if`.

File: tests/unit/test_utils.py [hunk 7]
```diff
@@ -76,35 +76,35 @@ def test_classify_field_ignores_case():
     assert expected == result
 
 
-def test_normalize_arxiv_category_returns_input_for_correct_category():
+def testnormalize_arxiv_category_returns_input_for_correct_category():
```
Comment (michamos) [inline tests/unit/test_utils.py]: why did you remove the `_` after `test` here and in the following tests? too eager search & replace I guess.

File: inspire_schemas/utils.py [hunk 8]
```diff
@@ -310,6 +310,18 @@
...
+RE_ARXIV_POST_2007_CLASS = re.compile(
+    "(arxiv:)?((?P<category>(?:[a-z\-]+)(?:\.[a-z]{2})?)/)?(?P<identifier>\d{4}\.\d{4,5})(v\d+)?$",
+    flags=re.I
+)
+
+RE_ARXIV_PRE_2007_CLASS = re.compile(
+    "(arxiv:)?(?P<category>(?P<extraidentifier>[a-z\-]+)(?:\.[a-z]{2})?)/(?P<identifier>\d{4}\d+)(v\d+)?$",
+    flags=re.I
+)
+
```
Comment (michamos) [inline inspire_schemas/utils.py]: Can you put this above, with the other regexes?

File: inspire_schemas/utils.py [hunk 9]
```diff
@@ -957,3 +969,43 @@ def normalize_isbn(isbn):
...
+    arxiv_test = obj.split()
+    if not arxiv_test:
+        return False
+
+    matched_arxiv = (RE_ARXIV_PRE_2007_CLASS.match(arxiv_test[0]) or
+                     RE_ARXIV_POST_2007_CLASS.match(arxiv_test[0]))
+
+    if not matched_arxiv:
+        return False
+    else:
```
Comment (michamos) [inline inspire_schemas/utils.py]: the whole point of this is to be able to avoid the `else`, to remove one indentation level.

File: tests/unit/test_utils.py [hunk 10]
```diff
@@ -850,3 +850,131 @@ def test_fix_reference_url_substitute_tilde():
...
+
+
+def test_normalize_arxiv_handles_new_identifiers_with_cls_prefix_and_ver():
+    expected = '1501.00001'
+    result = utils.normalize_arxiv('arXiv:hep-th.GT/1501.00001v1')
+
+    assert expected == result
+
+
+def text_normalize_arxiv_handles_uppercase():
```
Comment (michamos) [inline tests/unit/test_utils.py]: * `test` instead of `text`
* can you add similar tests for `is_arxiv`, also with negative examples?

File: tests/unit/test_utils.py [hunk 11]
```diff
@@ -850,3 +850,131 @@ def test_fix_reference_url_substitute_tilde():
...
+    expected = 'math/0312059'
+    result = utils.normalize_arxiv('MATH/0312059')
+
+    assert expected == result
+
+
+def text_normalize_arxiv_handles_uppercase_with_cls():
```
Comment (michamos) [inline tests/unit/test_utils.py]: same
